### PR TITLE
add-testing for controller

### PR DESCRIPTION
**Pull request by ANTOINE1128**
# Project requirements
## Add RSpec as a gem to your project.
 Create a Request spec file for all your controllers.
Make sure that for each action you check:
If response status was correct.
If a correct template was rendered.
If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/controllers/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      get "/users/#{@user.id}/posts"
+      expect(response.body).to include('This is the list of posts')
+    end
+
+    it 'should render the show action correctly' do
+      get "/users/#{@user.id}/posts/1"
+      expect(response).to be_successful
+      expect(response).to render_template('posts/show', layout: 'layouts/application')
+      expect(response.body).to include('this is your detail page')
+    end
```
Comment (codepantha) [inline spec/controllers/posts_controller_spec.rb]: - [ ] The two highlighted tests above are failing as you can see in the attached image below 👇 Please try to make them pass 👍 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @ANTOINE1128 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added controller specs 📌 
✔️ Good commit history
✔️ No linter errors 🟢 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers/posts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      expect(response).to render_template('posts/index')
+      expect(response).to render_template(layout: 'layouts/application')
+    end
+
+    it 'should render the index content correctly' do
+      get "/users/#{@user.id}/posts"
+      expect(response.body).to include('user page')
+    end
+  end
+end
```
Comment (julie-ify) [inline spec/controllers/posts_controller_spec.rb]: - [ ] You should have tests for the post index and show pages. You have removed post show tests that were present during the previous review. You were asked to fix the failing tests, not remove them. Kindly add the missing tests and ensure they are passing.

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @ANTOINE1128,

Good job so far!

## Highlights!
- Good code organization and tests ✔️
- Descriptive README ✔️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @ANTOINE1128 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Implemented tests for controllers ✔️ 
All tests passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**